### ISSUE TITLE
Issue regarding the environment.yaml

### ISSUE THREAD

**Issue by Jay-1409** [the-sky001/WiDistill#2]
Hey i think that the environment.yaml is specific to some particular os (i am guessing it was linux) and hence it throws errors in other operating systems.

**Comment by Shadi-Attarha** [comment]
Hey, I followed the steps as you described in the README. But after running the distill.py, the loss is always inf/nan with different learning rates.

**Comment by Jay-1409** [comment]
> Hey, I followed the steps as you described in the README. But after running the distill.py, the loss is always inf/nan with different learning rates.

Hey could you please tell me which dataset did you use?

**Comment by the-sky001** [comment]
Hi Shadi,

Thank you for your feedback and for trying out our code!

For the WiDar dataset, please note that we first preprocess the raw CSI data into the body-coordinate velocity profile (BVP) format, using the same BVP procedure as described by the official WiDar team. Only the BVP-processed data is compatible with our code. If you use the raw CSI directly, you will almost always get inf or nan loss values.

For the other datasets we used in the paper, BVP preprocessing is not required—you can use them as provided.

If you’re working with WiDar, please refer to the following for the BVP method:

Paper: Zero-Effort Cross-Domain Gesture Recognition with Wi-Fi
URL: https://ieeexplore.ieee.org/document/9148993

Alternatively, you can directly download the BVP-processed data from:
Widar3.0 WiFi-based Activity Recognition Dataset
URL: https://ieee-dataport.org/open-access/widar-30-wifi-based-activity-recognition-dataset

Sorry for not making this clear in the original README. If you have any other questions or need help with preprocessing, please feel free to ask!

Best,
Tiantian Wang

**Comment by the-sky001** [comment]
> Hey i think that the environment.yaml is specific to some particular os (i am guessing it was linux) and hence it throws errors in other operating systems.

Hi,

Yes, you are correct. The provided environment.yaml and code are currently only tested on Linux systems. You may encounter errors on other operating systems, and some environment settings may need to be modified manually.

For best results, please try to use a Linux environment.

Thank you for your understanding!

Best,
Tiantian Wang

**Comment by Jay-1409** [comment]
> Hi Shadi,
> 
> Thank you for your feedback and for trying out our code!
> 
> For the WiDar dataset, please note that we first preprocess the raw CSI data into the body-coordinate velocity profile (BVP) format, using the same BVP procedure as described by the official WiDar team. Only the BVP-processed data is compatible with our code. If you use the raw CSI directly, you will almost always get inf or nan loss values.
> 
> For the other datasets we used in the paper, BVP preprocessing is not required—you can use them as provided.
> 
> If you’re working with WiDar, please refer to the following for the BVP method:
> 
> Paper: Zero-Effort Cross-Domain Gesture Recognition with Wi-Fi
> URL: https://ieeexplore.ieee.org/document/9148993
> 
> Alternatively, you can directly download the BVP-processed data from:
> Widar3.0 WiFi-based Activity Recognition Dataset
> URL: https://ieee-dataport.org/open-access/widar-30-wifi-based-activity-recognition-dataset
> 
> Sorry for not making this clear in the original README. If you have any other questions or need help with preprocessing, please feel free to ask!
> 
> Best,
> Tiantian Wang
 
Hello 
Are there any preprocessing requirements for the mmfi dataset? 
Thank you so much :)

**Comment by the-sky001** [comment]
Hi, thanks for your question! For the mmfi dataset, some preprocessing is needed before training. The original data for each sample is shaped (3, 114, 10), where 3 is the number of antennas, 114 is the number of subcarriers, and 10 is the number of time samples within a 100ms window. There are 297 such files (frames).

<img width="1068" height="74" alt="Image" src="https://github.com/user-attachments/assets/24dd4acf-3c86-400f-a879-fc505804a318" />

<img width="1098" height="239" alt="Image" src="https://github.com/user-attachments/assets/99055644-4c99-4b7f-bfff-97d0169595aa" />
I combined the last two dimensions, so each frame becomes (342, 10) with 342 = 3 × 114. Using the segment file provided with the dataset (which defines action boundaries, e.g., 1-24, 25-50, etc.), I extracted a fixed-length sequence for each action.

<img width="1027" height="196" alt="Image" src="https://github.com/user-attachments/assets/24e52f93-2a4e-4ee2-8b7e-cd1028516d6d" />
Specifically, I took 35 frames per action segment; if the segment was longer, I truncated it, and if it was shorter, I padded with zeros. Then, I flattened the time window (10 samples per frame × 35 frames) into the temporal dimension, resulting in a final data shape of (342, 350) for each sample, where 350 = 35 × 10. Sorry for not explaining this preprocessing clearly before! For more details, please refer to the paper MM-Fi: Multi-Modal Non-Intrusive 4D Human Dataset for Versatile Wireless Sensing. If you have any more questions or need help with code, feel free to ask.

**Comment by Jay-1409** [comment]
> Hi, thanks for your question! For the mmfi dataset, some preprocessing is needed before training. The original data for each sample is shaped (3, 114, 10), where 3 is the number of antennas, 114 is the number of subcarriers, and 10 is the number of time samples within a 100ms window. There are 297 such files (frames).
> 
> <img width="1068" height="74" alt="Image" src="https://github.com/user-attachments/assets/24dd4acf-3c86-400f-a879-fc505804a318" />
> 
> <img width="1098" height="239" alt="Image" src="https://github.com/user-attachments/assets/99055644-4c99-4b7f-bfff-97d0169595aa" />
> I combined the last two dimensions, so each frame becomes (342, 10) with 342 = 3 × 114. Using the segment file provided with the dataset (which defines action boundaries, e.g., 1-24, 25-50, etc.), I extracted a fixed-length sequence for each action.
> 
> <img width="1027" height="196" alt="Image" src="https://github.com/user-attachments/assets/24e52f93-2a4e-4ee2-8b7e-cd1028516d6d" />
> Specifically, I took 35 frames per action segment; if the segment was longer, I truncated it, and if it was shorter, I padded with zeros. Then, I flattened the time window (10 samples per frame × 35 frames) into the temporal dimension, resulting in a final data shape of (342, 350) for each sample, where 350 = 35 × 10. Sorry for not explaining this preprocessing clearly before! For more details, please refer to the paper MM-Fi: Multi-Modal Non-Intrusive 4D Human Dataset for Versatile Wireless Sensing. If you have any more questions or need help with code, feel free to ask.

Yes!! 
I did ran into this particular error of shape mismatch while trying to visualise (running distill.py) using white and black.

I'll fix it :)

Thanks a lot!!

**Comment by Jay-1409** [comment]
One more thing @the-sky001 
What are the **system specifications** required? 
- When I try to run the distill.py, it ends up **flooding the RAM (16 GB)** and my system ultimately crashes!!
- could it be some issue created due to the teaks i have made to the code or the system requirnments exceed what i have.

Thank you so much!!

**Comment by the-sky001** [comment]
> One more thing [@the-sky001](https://github.com/the-sky001) What are the **system specifications** required?
> 
> * When I try to run the distill.py, it ends up **flooding the RAM (16 GB)** and my system ultimately crashes!!
> * could it be some issue created due to the teaks i have made to the code or the system requirnments exceed what i have.
> 
> Thank you so much!!

Hi, thanks for bringing this up!

For reference, my experiments were run on a workstation with over 100GB of RAM. The current implementation is indeed quite memory-intensive, and it’s not uncommon for machines with 16GB RAM to crash. So what you experienced is expected given the hardware differences.

If you made some changes to the code, it’s also possible that those tweaks increased the memory footprint further. However, even with the original version, handling the full dataset at once typically requires significantly more than 16GB RAM.

If you want to work on a system with limited memory, I recommend modifying the code to process data, or loading only a subset of the dataset into memory at a time, rather than the entire dataset. Alternatively, running the code on a machine with more RAM would also resolve the issue.

Sorry for the inconvenience, and thank you for your feedback!
